rust/src/lib/ip.rs

```diff
@@ -1245,23 +1245,73 @@ fn get_ip_prop_list(
-    Ok(format!(
-        "{}/{}",
-        new_ip_net.network(),
-        new_ip_net.prefix_len()
-    ))
-}
-
-fn ip_addr_to_ip_network(ip_addr: &str) -> String {
-    if !ip_addr.contains('/') {
-        if is_ipv6_addr(ip_addr) {
-            format!("{ip_addr}/{IPV6_ADDR_LEN}")
-        } else {
-            format!("{ip_addr}/{IPV4_ADDR_LEN}")
+    let ip_nets: Vec<&str> = ip_net.split('/').collect();
+    match ip_nets.len() {
+        0 => Err(NmstateError::new(
+            ErrorKind::InvalidArgument,
+            "Invalid IP network string, got empty string".into(),
+        )),
+        1 => {
+            let ip = IpAddr::from_str(ip_nets[0]).map_err(|e| {
+                NmstateError::new(
+                    ErrorKind::InvalidArgument,
+                    format!("Invalid IP network '{ip_net}': {e}",),
+                )
+            })?;
+            Ok(if ip.is_ipv6() {
+                format!("{ip}/{IPV6_ADDR_LEN}")
+            } else {
+                format!("{ip}/{IPV4_ADDR_LEN}")
+            })
         }
-    } else {
-        ip_addr.to_string()
+        2 => {
+            let prefix_len = ip_nets[1].parse::<usize>().map_err(|e| {
+                NmstateError::new(
+                    ErrorKind::InvalidArgument,
+                    format!(
+                        "Invalid IP network prefix length \
+                        '{}' in '{ip_net}': {e}",
+                        ip_nets[1]
+                    ),
+                )
+            })?;
+            let ip =
+                apply_ip_prefix_len(IpAddr::from_str(ip_nets[0])?, prefix_len);
+            if ip.is_ipv6() {
+                if prefix_len > IPV6_ADDR_LEN {
+                    Err(NmstateError::new(
+                        ErrorKind::InvalidArgument,
+                        format!(
+                            "Invalid IPv6 network prefix length '{}' \
+                            in '{ip_net}', should be smaller than \
+                            {IPV6_ADDR_LEN}'",
+                            ip_nets[1],
+                        ),
+                    ))
+                } else {
+                    Ok(format!("{ip}/{prefix_len}"))
+                }
+            } else if prefix_len > IPV4_ADDR_LEN {
+                Err(NmstateError::new(
+                    ErrorKind::InvalidArgument,
+                    format!(
+                        "Invalid IPv4 network prefix length '{}' \
+                        in '{ip_net}', should be smaller than \
+                        {IPV4_ADDR_LEN}'",
+                        ip_nets[1],
+                    ),
+                ))
+            } else {
+                Ok(format!("{ip}/{prefix_len}"))
+            }
+        }
+        _ => Err(NmstateError::new(
+            ErrorKind::InvalidArgument,
+            format!(
+                "Invalid IP network string: '{ip_net}', expecting \
+                    192.0.2.0/24 or 2001:db8:1::/64 or 192.0.2.1"
```

Comment:
are these examples ?

Reply:
Yes

Reply:
Changed to

```
+                "Invalid IP network string: '{ip_net}', expecting 'ip/prefix' \
+                format, for example: 192.0.2.0/24 or 2001:db8:1::/64 or \
+                192.0.2.1"
```

Reply:
perfect, looks better IMO. 

Thanks @ffmancera for making the concern I had more explicit :)